core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala

```diff
@@ -202,13 +202,23 @@ private[spark] class EventLoggingListener(
       }
     }
     fileSystem.rename(new Path(logPath + IN_PROGRESS), target)
+    writer = None
   }
 
+  Runtime.getRuntime.addShutdownHook(new Thread("Renaming inprogress log to logPath") {
+    override def run(): Unit = Utils.logUncaughtExceptions {
+      if (writer != None && fileSystem.exists(new Path(logPath + IN_PROGRESS))) {
+        logDebug("Inprogressing event log exists. Application may be terminated abnormally.")
+        fileSystem.rename(new Path(logPath + IN_PROGRESS), new Path(logPath + ABNORMAL))
```

Comment:
@viirya , I encountered exception with this `rename` operation on, and no exception without this line, did you meet this? I haven't figure out why.
```java.lang.IllegalArgumentException: Codec [abnormal] is not available. Consider setting spark.io.compression.codec=lzf
	at org.apache.spark.io.CompressionCodec$$anonfun$createCodec$1.apply(CompressionCodec.scala:73)
	at org.apache.spark.io.CompressionCodec$$anonfun$createCodec$1.apply(CompressionCodec.scala:73)
	at scala.Option.getOrElse(Option.scala:120)
	at org.apache.spark.io.CompressionCodec$.createCodec(CompressionCodec.scala:73)
	at org.apache.spark.scheduler.EventLoggingListener$$anonfun$8$$anonfun$apply$2.apply(EventLoggingListener.scala:287)
	at org.apache.spark.scheduler.EventLoggingListener$$anonfun$8$$anonfun$apply$2.apply(EventLoggingListener.scala:287)
	at scala.collection.mutable.MapLike$class.getOrElseUpdate(MapLike.scala:189)
	at scala.collection.mutable.AbstractMap.getOrElseUpdate(Map.scala:91)
	at org.apache.spark.scheduler.EventLoggingListener$$anonfun$8.apply(EventLoggingListener.scala:287)
	at org.apache.spark.scheduler.EventLoggingListener$$anonfun$8.apply(EventLoggingListener.scala:286)
	at scala.Option.map(Option.scala:145)
	at org.apache.spark.scheduler.EventLoggingListener$.openEventLog(EventLoggingListener.scala:286)
	at org.apache.spark.deploy.master.Master.rebuildSparkUI(Master.scala:773)
	at org.apache.spark.deploy.master.Master.removeApplication(Master.scala:710)
	at org.apache.spark.deploy.master.Master.finishApplication(Master.scala:688)```